components/lsp-form/LongShort.tsx

```diff
@@ -31,22 +40,40 @@ const LongShort: FC<Props> = ({
           placeholder="0.0"
           value={longTokenAmount}
           setValue={setLongTokenAmount}
+          additionalEffects={(e) => {
+            // Note: There is an effect that will cause the other input to update.
+            const newAmount =
+              Number(e.target.value) * Number(collateralPerPair);
+            setAmount(newAmount.toString());
+          }}
         />
         <TextInput
           label="short token"
           labelPlacement={"overlap" as LabelPlacement}
           placeholder="0.0"
           value={shortTokenAmount}
           setValue={setShortTokenAmount}
+          additionalEffects={(e) => {
+            // Note: There is an effect that will cause the other input to update.
+            const newAmount =
+              Number(e.target.value) * Number(collateralPerPair);
+            setAmount(newAmount.toString());
+          }}
         />
       </FormRow>
       <BalanceRowToken>
         <div>
-          <span>Your Balance 123.45 </span>
+          <span>
+            Your Balance{" "}
+            {ethers.utils.formatUnits(tokensMinted.toString(), erc20Decimals)}{" "}
```

Comment:
the label says balance, but you are showing tokens minted. just want to be sure what the intent here is, is it to show the users balance of long or short tokens? or is it to show how many have been minted through the interface? While for minting, i think we can assume a user will always mint equal amounts of long/short tokens. but their balance may be completely different, ie they may have more longs than shorts, since they can be traded on open market.

Reply:
Updated to proper balance values of contracts.